### DISCUSSION TITLE
is it possible to message control sequence queries without using stdin stderr?

### DISCUSSION THREAD

**Discussion by iambumblehead** [kovidgoyal/kitty#7215] [Q&A] [answered]
My script uses  this `read` pattern to send all control sequence queries, for example, a window size query,
```bash
IFS=";" read -a winsize -s -t 2 -d "t" -p $'\e[14t' >&2
echo "window resolution ${winsize[2]} ${winsize[1]}" # window resolution 696 469
```

problem: the above query returns empty values when the script runs inside `vifm` ncurses filemanager. Apparently, `ncurses` makes `stdin` and `stderr` un-available to the script.

I tried many ways to query window size from inside vifm and totally failed but found one way that almost works,
```bash
echo -e '\e[14t' > /dev/tty
IFS=";" read -a winsize -s -t 2 -d "t" < /dev/tty
echo "window resolution ${winsize[2]} ${winsize[1]}"
```
Window size is returned but the operation interferes with vifm. Junk characters appear in the screen and again later when entering input mode.

Please recommend a safe alternative way to send control sequence queries from inside ncurses, so they will not interfere with vifm.

Thanks in advance for any suggestions or advice.

**Comment by kovidgoyal** [comment, marked as answer]
What does this have to do with kitty? You should ask the nvim developers
how to do this. IIRC nvim has some channel mechanism you can use but I
dont recall the details as I dont use nvim.

And just FYI, you dont need to use escape codes to query screen size
https://sw.kovidgoyal.net/kitty/graphics-protocol/#getting-the-window-size

If you want to control kitty, you dont need to use the terminal
device you can setup remote control with a socket and use that,
regardless of nvim.
